### PR TITLE
6648 organization management openapi generator rework

### PR DESCRIPTION
**Pull request by TeroOtavia**
Refactoring organization management and user endpoints to use newly generated api from swagger file. Contains interface replace changes and some minor code logic refactoring to get everything to work with new system.

Resolves: #6648

### CODE REVIEW

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/user-index.ts [hunk 1]
```diff
@@ -331,15 +141,15 @@ export const userIndex: Reducer<UserIndexType> = (
  * @param action action
  */
 /* export default function userIndex(
```
Comment (avukkonen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/user-index.ts:143]: commented code

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/users/index.ts [hunk 2]
```diff
@@ -388,13 +382,13 @@ export const userSelect: Reducer<UsersSelectType> = (
  * @param action action
  */
 /* export function userSelect(
```
Comment (avukkonen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/users/index.ts:384]: commented code

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/users/index.ts [hunk 3]
```diff
@@ -297,7 +291,7 @@ export const userGroups: Reducer<UserGroupsType> = (
  * @param action action
  */
 /* export function userGroups(
```
Comment (avukkonen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/users/index.ts:293]: commented code

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/users/index.ts [hunk 4]
```diff
@@ -202,7 +196,7 @@ export const organizationUsers: Reducer<UsersType> = (
  * @param action action
  */
 /* export default function users(
```
Comment (avukkonen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/users/index.ts:198]: Commented code

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/base/status.ts [hunk 5]
```diff
@@ -131,16 +67,6 @@ export interface ProfileStatusType {
...
-  TEACHER = "TEACHER",
-  STUDENT = "STUDENT",
-  ADMINISTRATOR = "ADMINISTRATOR",
-  MANAGER = "MANAGER",
-  STUDY_PROGRAMME_LEADER = "STUDY_PROGRAMME_LEADER",
-  STUDY_GUIDER = "STUDY_GUIDER",
-  CUSTOM = "CUSTOM",
-}
-
 /* const workspaceIdNode = document.querySelector(
```
Comment (avukkonen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/base/status.ts:70]: commented code

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/guider/index.ts [hunk 6]
```diff
@@ -444,9 +435,7 @@ export const guider: Reducer<GuiderType> = (
 
     case "UPDATE_ONE_GUIDER_LABEL_FROM_ALL_STUDENTS": {
       // eslint-disable-next-line jsdoc/require-jsdoc
-      const mapFnStudentLabel = function (
-        label: GuiderStudentUserProfileLabelType
-      ) {
+      const mapFnStudentLabel = function (label: UserStudentFlag) {
```
Comment (avukkonen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/guider/index.ts:438]: jsdoc missing

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/guider/index.ts [hunk 7]
```diff
@@ -476,9 +465,7 @@ export const guider: Reducer<GuiderType> = (
 
     case "DELETE_ONE_GUIDER_LABEL_FROM_ALL_STUDENTS": {
       // eslint-disable-next-line jsdoc/require-jsdoc
-      const filterFnStudentLabel = function (
-        label: GuiderStudentUserProfileLabelType
-      ) {
+      const filterFnStudentLabel = function (label: UserStudentFlag) {
```
Comment (avukkonen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/guider/index.ts:468]: jsdoc missing

**Review by avukkonen — CHANGES_REQUESTED** [review]

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/user-index.ts (hunk 1 above)
Comment (TeroOtavia) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/user-index.ts:143]: Will be removed in its own issue

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/users/index.ts (hunk 2 above)
Comment (TeroOtavia) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/users/index.ts:384]: Will be removed in its own issue

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/users/index.ts (hunk 3 above)
Comment (TeroOtavia) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/users/index.ts:293]: Will be removed in its own issue

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/users/index.ts (hunk 4 above)
Comment (TeroOtavia) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/users/index.ts:198]: Will be removed in its own issue

**Review by avukkonen — APPROVED** [review]

**Pull request merged**